File: go/cmd/server/models/healthcheck.go
package models

type HealthCheckExecutorType string

const (
	DEFAULT HealthCheckExecutorType = "default"
	CURL    HealthCheckExecutorType = "curl"
	FETCH   HealthCheckExecutorType = "fetch"
)

type HealthCheck struct {
	Name                 string                  `json:"name"`
	Description          string                  `json:"description,omitempty"`
	URL                  string                  `json:"url"`
	Interval             int                     `json:"interval"`
	Method               string                  `json:"method,omitempty"`
	ExpectedResponseCode int                     `json:"expectedResponseCode,omitempty"`
	Timeout              int                     `json:"timeout,omitempty"`
	Active               bool                    `json:"active,omitempty"`
	Executor             HealthCheckExecutorType `json:"executor,omitempty"`
	Auth                 struct {
		Username string `json:"username"`
		Password string `json:"password"`
		APIKey   string `json:"apiKey"`
	} `json:"auth,omitempty"`
}

File: ui/src/types/healthcheck.ts
export interface Model {
  id: string
}

export enum HealthCheckStatus {
  HEALTHY = 'healthy',
  UNHEALTHY = 'unhealthy',
  DEGRADED = 'degraded'
}

export enum HealthCheckExecutorType {
  DEFAULT = 'default',
  CURL = 'curl',
  FETCH = 'fetch'
}

export type HealthCheckResponse = {
  status: HealthCheckStatus
}

export interface HealthCheck extends Model {
  id: string
  name: string
  url: string
  description?: string
  active?: boolean
  interval: number
  method?: string
  expectedResponseCode?: number
  timeout?: number
  auth?: {
    username: string
    password: string
    apiKey: string
  }
  executor?: HealthCheckExecutorType
}
